src/main/resources/assets/computercraft/lua/rom/programs/shell

```diff
@@ -180,7 +180,11 @@ function shell.programs( _bIncludeHidden )
 	-- Sort and return
 	local tItemList = {}
 	for sItem, b in pairs( tItems ) do
-		table.insert( tItemList, sItem )
+	    if sItem:find(".lua") == sItem:len()-3 then
```

Comment:
This won't work for files with multiple `.lua`s in them. Instead, do `if sItem:find("%.lua$) then` or `if sItem:sub(-4, -1) == ".lua" then`.

I'd also add a check that the file without the `.lua` extension isn't also in the `tItems` set.

This bit of code uses tabs, so you might want to use them here too. I'd really recommend turning on "show invisibles/whitespace" in your editor so you can see this better.